### ISSUE TITLE
Package names with periods don't work (e.g. "@mylib/a.b")

### ISSUE THREAD

**Issue by bmcbarron** [0x80/isolate-package#16]
In [create-packages-registry.ts](https://github.com/0x80/isolate-package/blob/6903f4c113a7c48603f40a5ead857b98a21fa0ce/src/helpers/create-packages-registry.ts#L94), the registry is constructed which indexes package manifests by their name. However, the registry object is built using [lodash.set](https://lodash.com/docs/4.17.15#set), which parses the key as a recursive path when it includes periods, so that package name "@mylib/a.b" results in `{ "@mylib/a": { "b": { ... }}}`. Later, when "@mylib/a.b" is looked up in the registry, it's not found resulting in the error `error TypeError: Cannot read properties of undefined (reading 'rootRelativeDir')`

It appears that the extra behavior of lodash.set is not needed. If so, the following change fixes this issue:
```typescript
.reduce<PackagesRegistry>(
    (acc, info) => {
      if (info) {
        acc[info.manifest.name] = info;
      }
      return acc;
    },
    {}
  );
```

**Comment by 0x80** [comment]
It wouldn't be difficult to fix this but I have never seen dots being used in Node.js package names. Could you point me to some documentation which describes this as being supported?

**Comment by bmcbarron** [comment]
[Here's the docs on `name`](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#name).
[Here's an excerpt about URL-safe characters](https://stackoverflow.com/a/695469/8107589).
Bottom line: dot, dash, underscore, tilde, and alphanumeric characters are all legal. Additional characters may work or they may not, depending on the particular tool and context.

I'm personally a fan of the [Robustness principle](https://en.wikipedia.org/wiki/Robustness_principle) which suggests that tools should be as relaxed as possible in what they accept.

**Comment by 0x80** [comment]
Fair enough! The NPM name rules are clearly less strict than I had assumed from experience. I have published v1.3.1 which contains this fix.
